zaza/openstack/charm_tests/cinder_lvm/tests.py

```diff
@@ -0,0 +1,136 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Encapsulate cinder-lvm testing."""
+
+import logging
+import uuid
+
+import zaza.model
+import zaza.openstack.charm_tests.test_utils as test_utils
+import zaza.openstack.utilities.openstack as openstack_utils
+
+
+def with_conf(application, config, model_name=None):
+    """Temporarily change the config options for an application in a model."""
+    prev = {}
+    for key in config.keys():
+        prev[key] = str(openstack_utils.get_application_config_option(
+            application, key, model_name=model_name))
+
+    def patched(f):
+        def inner(*args, **kwargs):
+            try:
+                zaza.model.set_application_config(
+                    application, config, model_name=model_name)
+                zaza.model.wait_for_agent_status(model_name=model_name)
+                return f(*args, **kwargs)
+            finally:
+                zaza.model.set_application_config(
+                    application, prev, model_name=model_name)
+                zaza.model.wait_for_agent_status(model_name=model_name)
+        return inner
+    return patched
+
+
+class CinderLVMTest(test_utils.OpenStackBaseTest):
+    """Encapsulate cinder-lvm tests."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Run class setup for running tests."""
+        super(CinderLVMTest, cls).setUpClass(application_name='cinder-lvm')
+        cls.model_name = zaza.model.get_juju_model()
+        cls.cinder_client = openstack_utils.get_cinder_session_client(
+            cls.keystone_session)
+
+    @classmethod
+    def tearDown(cls):
+        """Remove test resources."""
+        volumes = cls.cinder_client.volumes
+        for volume in volumes.list():
+            if volume.name.startswith('zaza'):
+                try:
+                    volume.detach()
+                    volumes.delete(volume)
+                except Exception:
+                    pass
+
+    def test_cinder_config(self):
+        """Test that configuration options match our expectations."""
+        logging.info('cinder-lvm')
+        expected_contents = {
+            'LVM-zaza-lvm': {
+                'volume_clear': ['zero'],
+                'volumes_dir': ['/var/lib/cinder/volumes'],
+                'volume_name_template': ['volume-%s'],
+                'volume_clear_size': ['0'],
+                'volume_driver': ['cinder.volume.drivers.lvm.LVMVolumeDriver'],
+            }}
+
+        zaza.model.run_on_leader(
+            'cinder',
+            'sudo cp /etc/cinder/cinder.conf /tmp/',
+            model_name=self.model_name)
```

Comment:
there is no need to pass the model_name unless you are doing something special, zaza internally uses `zaza.model.run_in_model()`[0] which will rely on `get_juju_model()` just like you do in `setUpClass()`.

[0] https://github.com/openstack-charmers/zaza/blob/master/zaza/model.py#L186